realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1269,6 +1279,37 @@ public static Object getDefaultModule() {
     }
 
     /**
+     * Acquire a reference to the given Realm file.
+     * This function call should be called in synchronized block on BaseRealm.class
+     */
+    private static void acquireRealmFileReference(RealmConfiguration configuration) {
+        String path = configuration.getPath();
+        Integer refCount = realmFileReferenceCounter.get(path);
+        if (refCount == null) {
+            refCount = 0;
+        }
+        realmFileReferenceCounter.put(path, refCount + 1);
```

Comment:
you should still call through to the parent method in BaseRealm, because it's counters also need to be accurate.